src/wire/Socket.ts

```diff
@@ -339,7 +341,8 @@ export class InteractiveSocket extends EventEmitter {
     }
 
     private sendRaw(packet: any) {
-        const data = JSON.stringify(packet);
+        // Replace circular references with nothing.
+        const data = stringify(packet, null, 0, () => {/** */});
```

Comment:
The only recursive data structure that is specified on the protocol is the memory warning structure the server sends. Sending any recursive packet is going to be invalid. I think that throwing an error is the right thing to do here.